src/components/Filter/Filter.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import classNames from 'classnames';
+import { TodoStatus } from '../../types/TodoStatus';
+
+type Props = {
+  counterActiveTodos: number;
+  counterCompletedTodos: number;
+  selectedFilter: TodoStatus;
+  onFilterSelect: (newFilter: TodoStatus) => void;
+  onClearCompleted: () => void;
+};
+
+const filterOptions = Object.values(TodoStatus);
+
+export const Filter: React.FC<Props> = React.memo(
+  ({
+    counterActiveTodos,
+    counterCompletedTodos,
+    selectedFilter,
+    onFilterSelect,
+    onClearCompleted,
+  }) => {
+    return (
+      <footer className="todoapp__footer">
+        <span className="todo-count">{`${counterActiveTodos} items left`}</span>
+
+        <nav className="filter">
+          {filterOptions.map((option) => (
+            <a
```

Comment:
make a separate component for it

Reply:
do it, split this with components

Reply:
i tried and caught this error `is not assignable to type 'intrinsicattributes & props`

Reply:
Bc you do something wrong :D